src/base/bittorrent/torrentinfo.cpp

```diff
@@ -213,24 +214,79 @@ QByteArray TorrentInfo::metadata() const
 
 QStringList TorrentInfo::filesForPiece(int pieceIndex) const
 {
-    if (pieceIndex < 0)
-        return QStringList();
+    // no checks here since fileIndicesForPiece() will return an empty list
+    QList<int> fileIndices = fileIndicesForPiece(pieceIndex);
+
+    QStringList res;
+    std::transform(fileIndices.begin(), fileIndices.end(), std::back_inserter(res), [this](int i) {
+        return this->filePath(i);
+    });
+
```

Comment:
Another point is that templates usually bloat the the file size. (again not saying you should change it)